crates/nu-protocol/src/config.rs

```diff
@@ -127,6 +128,7 @@ impl Default for Config {
             show_clickable_links_in_ls: true,
             render_right_prompt_on_last_line: false,
             explore: HashMap::new(),
+            recursion: 50,
```

Comment:
I see where this is being defaulted to 50 in the config.rs but where is it being read from the config file? Or is this not meant to be configurable and it's just meant to start out with a default?

Reply:
In `engine_state` which is initial the process of `eval` and available in all functions .
on `engine_state` there is a `config` part on struct which call `Default` on `Config` and it will initialize the value.
As this is necessary to have `engine_state` , it will initialize 100%.

Reply:
Right. What I'm saying is that you're adding a setting to config, recursion, that is hard coded with no way of setting it via config.nu, right?

Also, maybe it should be named `recursion_limit` or something that indicates what it's for.

Reply:
Can we delete it then ? 
Just use recursion limit which rust allow us , that is 128 ...
Because I think it is unnecessary

Reply:
@fdncred 
I added it to `config.nu` and its default its 50.
parameter name changed to recursion_limit too .

Reply:
Sorry for the late reply. I'm fine with using whatever rust's default recursion limit is and removing/not having a configuration point for this. I'm not sure how much this configuration point would ever be used.

Reply:
Deleted from Config . @fdncred